builders/tutorials/subsquid.md

```diff
@@ -0,0 +1,428 @@
+
+1. The job of `main.ts` is to run the processor and refine the collected data. In `processor.run`, the processor will iterate through all selected blocks and look for transfer event logs. Whenever it finds a transfer event, it's going to store it in an array of transfer events where it awaits further processing. 
+
+2. The `transferEvent` interface is the type of structure that stores the data extracted from the event logs.
+
+3. `getTransfer` is a helper function that extracts and decodes ERC-20 transfer event data from a log entry. It constructs and returns a TransferEvent object, which includes details such as the transaction ID, block number, sender and receiver addresses, and the amount transferred. `getTransfer` is called at the time of storing the relevant transfer events into the array of transfers.
+
+4. `processTransfers` enriches the transfer data and then inserts these records into a typeorm database using the `ctx.store` methods. The account model, while not strictly necessary, allows us to introduce another entity in the schema to demonstrate working with multiple entities in your Squid.
+
+5. `getAccount` is a helper function that manages the retrieval and creation of account objects. Given an account ID and a map of existing accounts, it returns the corresponding account object. If the account doesn't exist in the map, it creates a new one, adds it to the map, and then returns it.
+
+We'll demo a sample query a later section. You can copy and paste the below code into your `main.ts` file:
+
+???+ code "main.ts"
+
+    ```ts
+    --8<-- 'code/tutorials/subsquid/main.ts'
+    ```
+
+Now we've taken all of the steps necessary and are ready to run our indexer!
+
+### Run the Indexer {: #run-indexer }
+
+To run our indexer, we're going to run a series of `sqd` commands:
+
+1. Build our project
+
+   ```bash
+   sqd build
+   ```
+2. Launch the database:
+
+   ```bash
+   sqd up
+   ```
+3. Remove the database migration file that comes with the EVM template and generate a new one for our new database schema:
+
+   ```bash
+   sqd migration:generate
+   sqd migration:apply
+   ```
+4. Launch the processor:
+
+   ```bash
+   sqd process
+   ```
+
+In your terminal, you should see your indexer starting to process blocks!
+
+![Get Squid running](/images/builders/tutorials/subsquid/subsquid-4.png)
+
+## Querying your Squid {: #querying-your-squid }
+
+To query your squid, open up a new terminal window within your project and run the following command:
+
+```bash
+sqd serve
+```
+
+And that's it! You can now run queries against your Squid on the GraphQL playground at [http://localhost:4350/graphql](http://localhost:4350/graphql){target=\_blank}. Try crafting your own GraphQL query, or use the below one:
+
+???+ code "sample-query.graphql"
```

Comment:
this isn't a file in the project, is it? if it is, then yeah you can keep it like this but if it's not you can use "Example query" or whatever